packages/react-ui/src/hooks/useImperativeIntl.js

```diff
@@ -0,0 +1,33 @@
+import { createIntl, createIntlCache } from 'react-intl';
+import { messages } from '../localization';
+import { useMemo } from 'react';
+
+const DEFAULT_LOCALE = 'en-US';
+const cache = createIntlCache();
+
+/**
+ * The hook returns an instance of the Intl object with the specified locale
+ * and messages using imperative API.
+ * @see https://formatjs.io/docs/react-intl/api/#createintl
+ * @param {string} locale - The locale to use.
+ * @returns the `intl` object.
+ */
+export default function useImperativeIntl(locale) {
+  // Check if the locale is valid ISO 5 letters code (ex: 'en-US', 'es-ES', 'id-ID')
+  const testLocale = /^[a-z]{2}-[A-Z]{2}$/;
+  const validLocale = testLocale.test(locale) ? locale : DEFAULT_LOCALE;
```

Comment:
Is it possible to check if the locale is the shortest and add the next part by default? If the locale is `en`, the full location is `en-EN` so we can build it.

Reply:
In my opinion it is better to pass the complete ISO since it is required for the formatting of dates and numbers, and the same format is not applied to es-ES as to es-MX or en-EN to en-US for example